sdk/typescript/src/utils/publish.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Base64DataBuffer } from '../serialization/base64';
+
+/**
+ * Use the sui Command line tool to verify, compile and compute
+ * the topological ordering of the module dependencies within
+ * the package. This method is only available in NodeJS environment.
+ *
+ * Use this method with the local-txn-data-serializer.
+ *
+ * @param packagePath path to the Move package to be published
+ * @param cliPath path to the sui CLI executable. Follow
+ * {@link https://docs.sui.io/devnet/build/install} to install
+ * @returns an array of compiled modules in bytes
+ */
+export function getCompiledModules(
+  packagePath: String,
+  cliPath: String = 'sui'
+): number[][] {
+  const { execSync } = require('child_process');
+  const modules = execSync(
+    // gas-budget is a placeholder since there's no on-chain execution and thus no gas cost
+    `${cliPath} client publish --path ${packagePath} --gas-budget 30000 --no-execute`,
```

Comment:
IMO we shouldn't depend on the global `sui` binary, we could package it into something like `@mysten/sui-bin` that is invoked instead.

Reply:
> we could package it into something like @mysten/sui-bin that is invoked instead

Agree that @mysten/sui-bin can be the fallback/default version, but I do think having the option to pass in the CLI binary path offers more flexibility. For example, the developers can point to the `target/debug/sui` locally that has not been released through `@mysten/sui-bin`

Reply:
Yeah, I'm also fine with letting this value be dynamic, more that by default it shouldn't even require installation of sui at all.